packages/base/src/panelview/formbuilder.tsx

```diff
@@ -202,8 +219,92 @@ export class ObjectPropertiesForm extends React.Component<IProps, IStates> {
           </div>
         </div>
       );
-    } else {
-      return <div>{this.buildForm()}</div>;
     }
   }
 }
+
+export class LayerPropertiesForm extends ObjectPropertiesForm {
+  protected processSourceData(sourceData: IDict<any>) {
+    // Replace the source id by its name in the form
+    sourceData.source = this.props.model.getSource(sourceData.source)?.name;
```

Comment:
We should probably know the layer type, to list only the related sources
- raster layer -> raster source
- fill or point layer -> geoJSON source or other data source

Reply:
Right! We should probably have an API to perform this in the `IJupyterGISModel`, something like: `getSourcesByType(type: string)`?